src/AMDGPU.h

```diff
@@ -40,13 +39,20 @@ namespace AMDGPU
 
     inline uint32_t GetTemperature()
     {
-        if (!RuntimeConfig::Get().hasAMD)
+        if (!RuntimeConfig::Get().hasAMD && Config::Get().gpuThermalZone.empty())
```

Comment:
The additional check shouldn't be here (The check is only intended to fail if AMDGPU is disabled and this method is still called, meaning a developer error)